File: ENTIDADES/Almacen/AGuiaSalida.cs
using ENTIDADES.Generales;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace ENTIDADES.Almacen
{
    [Table("GuiaSalida", Schema = "Almacen")]
  public  class AGuiaSalida : AuditoriaColumna
    {
	
        [Key]
        public long idguiasalida { get; set; }
        public string codigo { get; set; }
        public DateTime fechatraslado { get; set; }
        public int idsucursal { get; set; }
        public int idalmacensucursalorigen { get; set; }
        public int? idalmacensucursaldestino { get; set; }
        public int idsucursaldestino { get; set; }
        public int idcaja { get; set; }
        public int idcorrelativo { get; set; }       
        public int idempresa { get; set; }        
        public string seriedoc { get; set; }        
        public string numdoc { get; set; }        
        public string estado { get; set; }        
        public string estadoguia { get; set; }        
        public int ano { get; set; }
        public string observacion { get; set; }
        public int? idempresatransporte { get; set; }
        public int? idvehiculo { get; set; }
        public string idempleadoaudita { get; set; }
        public string idempleadomantenimiento { get; set; }
        public int? idproveedor { get; set; }
        public int? idtipoguia { get; set; }
        public int? idventa { get; set; }
        public decimal? peso_total { get; set; }
        public decimal? bultos { get; set; }

        [ForeignKey("idempresa")]
        public Empresa empresa { get; set; }
        [ForeignKey("idsucursal")]
        public SUCURSAL sucursal { get; set; }
        [NotMapped]
        public EMPLEADO empleado { get; set; }
        [NotMapped]
        public string jsondetalle { get; set; }

        //variables notmapped
        [NotMapped]
        public string encargado  { get; set; }
        [NotMapped]
        public string sucursaldestino  { get; set; }
        public int? IdDistribucion { get; set; }
        public int? idcliente { get; set; }
    }
}

File: ERP/wwwroot/js/modelos/almacen/AGuiaSalida.js

class AGuiaSalida {   
    constructor() {
        this.idguiasalida= 0;
        this.codigo = '';
        this.fechatraslado= '';
        this.idsucursal= 0;
        this.idalmacensucursalorigen= 0;
        this.idalmacensucursaldestino= 0;
        this.idsucursaldestino= 0;
        this.idcaja= 0;
        this.idcorrelativo= 0;
        this.idempresa= 0;
        this.seriedoc= '';
        this.numdoc= '';
        this.estado= '';
        this.estadoguia= '';
        this.ano= 0;
        this.observacion= '';
        this.idempresatransporte = 0;
        this.idvehiculo= 0;
        this.idempleadoaudita= 0;
        this.idempleadomantenimiento = '';
        this.idproveedor = 0;
        this.idcliente = 0;
        this.idtipoguia = 0;
        this.tieneguia = 0;
        this.empresa= '';
        this.sucursal= '';
        this.empleado= '';
        this.jsondetalle = '';
        //campos no mapeables
        this.encargado = '';
        this.sucursaldestino = '';
        this.proveedor = '';
    }
}

class ADetalleGuiaSalida {
    constructor() {
        this.iddetalleguiasalida = 0;
        this.idproducto= 0;
        this.idstock= 0;
        this.cantidadgenerada= 0;
        this.cantidadpicking= 0;
        this.estado= '';
        this.idguiasalida = 0;
        //campos no mapeables
        this.idsucusaldestino = 0;
        this.idproveedor = 0;
        this.producto = '';
        this.idalmacensucursalorigen = 0;
        this.lotecliente = 0;
        this.fechavencimientocliente = 0;
    }   
}